handler/README.md

```diff
@@ -1,232 +1,378 @@
+```
+F -> (F', T)
+```
+where `F` is Fragment, `F'` is modified Fragment, `T` is Transition.
+
+The node definition is abstract. It allows to define simple processing nodes but also more complex 
+structures such as a list of subgraphs. Furthermore, such a definition inspires to provide custom 
+node implementations.
+
+The `node` configuration is simple:
+```hocon
+graph {
+  node {
+    factory = action
     config {
-      endpointOptions {
-        path = /product/id
-        domain = webapi
-        port = 8080
-        allowedRequestHeaders = ["Content-Type"]
-      }
+      action = book-rest-api
     }
   }
-  # simple action with fallback
-  product-fallback {
-    factory = "inline-body"
+}
+
+
+```
+The `factory` parameter specifies a node factory name, `config` contains all options passed to 
+the factory. 
+
+Knot.x provides two node implementations:
+**Action node** that represents simple steps in a graph such as integration with a data source
+**SubTasks node** that is a list of unnamed tasks (sub-tasks) that are evaluated in parallel
+
+##### Action node
+Action node declares [Action](#actions) to execute by action name. Its configuration is simple:
+```hocon
+graph {
+  node {
+    factory = action
     config {
-      body = <div>Product not available at the moment</div>
+      action = book-rest-api
     }
   }
 }
 ```
+The above example specifies the action node that delegates processing to the `book-rest-api` action.
 
-## Task
-Task defines a directed Actions graph. Task definition (JSON) is transformed to graph of nodes: 
-- [ActionNode](https://github.com/Knotx/knotx-fragments/blob/master/handler/engine/src/main/java/io/knotx/fragments/engine/graph/ActionNode.java)
-- [CompositeNode](https://github.com/Knotx/knotx-fragments/blob/master/handler/engine/src/main/java/io/knotx/fragments/engine/graph/CompositeNode.java). 
-
-The example `pdp` task defines two action nodes. 
-
-A composite node is defined as a subgraph:
-```
-book-and-author {
-  actions = [
-    { action = book },
-    { action = author }
-  ]
-  onTransition {
-    _error {
-      action = book-and-author-fallback
+Knot.x allows simplifying action nodes declaration:
+```hocon
+action = book-rest-api
+```
+
+So the task definition is:
+```hocon
+tasks {
+  myTask {
+    action = book-rest-api
+  }  
+}
+```
+It is the `myTask` task with the action node using the `book-rest-api` action and no transitions.
+
+A nice syntax sugar!
+
+##### SubTasks node
+SubTasks node is a node containing a list of sub-tasks. It evaluates all of them sequentially. However, 
+all operations are non-blocking so they are executed in parallel. 
+
+Moreover, a list of sub-tasks must fit the `F -> (F',T)` function. Each subtask has its fragment 
+context, execute it's logic and update fragment payload (its own copy). Finally, when all sub-tasks are completed, 
+all payloads are merged and the new Fragment is returned.
+
+> Note that body modifications are not allowed.
```

Comment:
```suggestion
> Note that body modifications are not allowed because of the parallel execution of subtask nodes and the final `body` form cannot be determined. However, updating the fragment's `payload` is fine since all subtask nodes have their unique namespaces.
```